openedx/core/djangoapps/discussions/views.py

```diff
@@ -61,4 +122,117 @@ def post(self, request, course_key_string: str, **_kwargs) -> Response:
+
+
+class DiscussionsProvidersView(APIView):
+    """
+    Handle configuration-related view-logic
+    """
+    authentication_classes = (
+        JwtAuthentication,
+        BearerAuthenticationAllowInactiveUser,
+        SessionAuthenticationAllowInactiveUser
+    )
+    permission_classes = (IsStaffOrCourseTeam,)
+
+    @apidocs.schema(
+        parameters=[
+            apidocs.string_parameter(
+                'course_id',
+                apidocs.ParameterLocation.PATH,
+                description="The course for which to get provider list",
+            )
+        ],
+        responses={
+            200: DiscussionsProvidersSerializer,
+            401: "The requester is not authenticated.",
+            403: "The requester cannot access the specified course.",
+            404: "The requested course does not exist.",
+        },
+    )
+    def get(self, request, course_key_string: str, **_kwargs) -> Response:
+        """
+        Handle HTTP/GET requests
+        """
+        data = self.get_provider_data(course_key_string)
+        return Response(data)
+
+    @staticmethod
+    def get_provider_data(course_key_string: str) -> Dict:
+        """
+        Get provider data for specified course
+        Args:
+            course_key_string (str): course key string
+
+        Returns:
+            Dict: course discussion providers
+        """
+        course_key = validate_course_key(course_key_string)
+        configuration = DiscussionsConfiguration.get(course_key)
+        hidden_providers = []
+        # If the user is currently using the legacy provider, don't show the new provider
+        # TODO: Allow switching between legacy and new providers
+        if configuration.provider_type == Provider.LEGACY:
+            hidden_providers.append(Provider.OPEN_EDX)
+        # If the user is currently using the new provider, don't show the legacy provider
+        elif configuration.provider_type == Provider.OPEN_EDX:
+            hidden_providers.append(Provider.LEGACY)
+        else:
+            # If this is a new course, or some other provider is selected, the new provider
+            # should only show up if the MFE is enabled
+            if not ENABLE_DISCUSSIONS_MFE.is_enabled(course_key):
+                hidden_providers.append(Provider.OPEN_EDX)
+        serializer = DiscussionsProvidersSerializer(
+            {
+                'features': [
+                    {'id': feature.value, 'feature_support_type': feature.feature_support_type}
+                    for feature in Features
+                ],
+                'active': configuration.provider_type,
+                'available': {
+                    key: value
+                    for key, value in AVAILABLE_PROVIDER_MAP.items()
+                    if key not in hidden_providers
+                },
+            }
+        )
+        return serializer.data
+
+
+class CombinedDiscussionsConfigurationView(DiscussionsConfigurationView):
```

Comment:
I think mainly it's written like this right now, so that we can have a smooth transition between the new and old APIs. Once the frontend supports the new API structure, and both are successfully deployed and working, we can remove this class and API endpoint.